port/unix/omrsysinfo.c

```diff
@@ -6719,41 +6821,62 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 			state->count += 1;
 			goto _end;
 		}
-
-		state->fileMetricCounter = 0;
 	}
 
 	if (state->fileMetricCounter < subsystemMetricMapElement->metricElementsCount) {
-		BOOLEAN isMetricFound = FALSE;
 		char *current = state->fileContent;
-		char *end = state->fileContent + strlen(state->fileContent) - 1;
-		int32_t metricKeyLen = 0;
 		currentElement += state->fileMetricCounter;
-		metricKeyLen = strlen(currentElement->metricKeyInFile);
-		while (current < end) {
-			if (0 == strncmp(current, currentElement->metricKeyInFile, metricKeyLen)) {
-				char *newLine = current;
-				current += metricKeyLen;
-				/* advance past any whitespace */
-				while ((current != end) &&  (' ' == *current)) {
+
+		if ((OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE))
+			&& (CGROUP_METRIC_NO_KEY == (uintptr_t)currentElement->metricKeyInFile)
+		) {
+			/* File contains space-separated values with no keys. */
+			char *valueEnd = NULL;
+			int32_t i = 0;
+			for (i = 0; i < state->fileMetricCounter; i++) {
+				/* Skip values that were already read. */
+				while (' ' != *current) {
 					current += 1;
 				}
-				/* find the newline */
-				newLine = current;
-				while ('\n' != *newLine) {
-					newLine += 1;
+				current += 1;
+			}
+			valueEnd = current;
+			/* Find the last char of the value. */
+			while ((' ' != *valueEnd) && ('\n' != *valueEnd)) {
+				valueEnd += 1;
+			}
+			strncpy(metricElement->value, current, valueEnd - current);
+			metricElement->value[valueEnd - current] = '\0';
```

Comment:
Shouldn't you set `rc` to `0` here?

Reply:
rc will always be 0 at this point but I should set it to 0 for consistency with the else block